statsmodels/regression/censored_model.py

```diff
@@ -0,0 +1,900 @@
+
+    @set_transform
+    def fit(self, start_params = None, method='newton', **kwargs):
+        """
+        Fit the Tobit model
+
+        Parameters
+        ----------
+
+        Notes
+        -----
+        It is not recommended to change the default solver or starting
+        parameters.
+        """
+        if start_params is None:
+            start_params = self._get_start_params()
+        else:
+            start_params = np.copy(start_params)
+
+        mlefit = super(Tobit, self).fit(start_params=start_params,
+                        method=method, **kwargs)
+        return TobitResults(self, mlefit)
+
+class TobitResults(base.LikelihoodModelResults):
+    def __init__(self, model, mlefit):
+        self.model = model
+        params = mlefit.params
+        # re-parameterize from exponential
+        params[-1] = np.exp(params[-1])
+        # re-parameterize from Olsen
+        #params = _reparam_olsen(params)
+        beta, sigma = params[:-1], params[-1]
+        self.model_params = beta
+        mlefit.params = np.r_[beta, sigma]
+        self.__dict__.update(mlefit.__dict__)
+        self.scale = sigma # overwrite from mlefit
+        self._model_stats() # attach model statistics
+
+    def _model_stats(self):
+        model = self.model
+        exog = model.exog
+        self.nobs = nobs = exog.shape[0]
+        self.n_lcens = model.n_lcens
+        self.n_rcens = model.n_rcens
+        self.n_ucens = nobs - self.n_lcens - self.n_rcens
+        self.k_constant = k_constant = int(np.any(exog.var(0) == 0))
+        self.k_params = k_params = exog.shape[1] - k_constant
+        self.k_ancillary = 1 # number of ancillary parameters, sigma here
+        self.df_model = k_params - self.k_constant + 1 # for sigma
+        self.df_resid = nobs - self.df_model
+
+    def cov_params(self):
+        hess = self.model.hessian(self.params)
+        return np.linalg.inv(-hess)
+
+    @cache_readonly
+    def chi2(self):
+        return 2 * (self.llf-self.llnull)
+
+    @cache_readonly
+    def chi2_pvalue(self):
+        from scipy.stats import chi2
+        #NOTE: df == number of restrictions
+        return chi2.sf(self.chi2, self.df_model)
+
+    @cache_readonly
+    def llnull(self):
+        endog = self.model.endog
+        left = self.model.left
+        right = self.model.right
+        res = Tobit(endog, np.ones_like(endog), left=left,
+                     right=right).fit(method='bfgs', disp=0)
+        try:
+            assert not res.mle_retvals['warnflag']
+        except:
+            raise AssertionError("Null Likelihood did not converge. Please "
+                    "report.")
+        return res.llf
+
+    def predict(self, exog=None, typ='xb'):
+        return self.model.predict(self.params, exog, typ)
+    #TODO: fix this to remove params, we need a general solution here
+    predict.__doc__ = Tobit.predict.__doc__
+
+def webuse(data, baseurl='http://www.stata-press.com/data/r11/'):
```

Comment:
outdated
I think we have/had a function like this merged, but it should also work with pandas directly

Reply:
returns structured dtype, recarray